src/modules/profile/profile.service.ts

```diff
@@ -11,6 +12,36 @@ export class ProfileService {
     @InjectRepository(User) private userRepository: Repository<User>
   ) {}
 
+  async createProfile(createProfileDto: CreateProfileDto, userId: string) {
+    try {
+      // Fetch the user entity by userId
+      const user = await this.userRepository.findOne({ where: { id: userId } });
+      if (!user) {
+        throw new BadRequestException('User not found');
+      }
+
+      // Check if a profile for this user already exists
+      const existingProfile = await this.profileRepository.findOne({ where: { id: userId } });
+      if (existingProfile) {
+        throw new BadRequestException('Profile already exists');
+      }
+      const profile = this.profileRepository.create({ ...createProfileDto, user_id: user });
+      await this.profileRepository.save(profile);
+
+      const newProfile = await this.profileRepository.findOne({ where: { id: profile.id } });
+
+      if (!newProfile) {
+        throw new BadRequestException('Profile creation unsuccessful');
+      }
+      return newProfile;
+    } catch (error) {
+      if (error instanceof NotFoundException || error instanceof BadRequestException) {
```

Comment:
no need for try/catch block as nest wraps a try catch across all routes, the exceptions layer. any non specific error will be thrown as internal server by default